push_to_holmes.go

```diff
@@ -14,8 +14,13 @@ import (
 	"net/http"
 	"os"
 	"time"
+	"path/filepath"
 
 	"gopkg.in/mgo.v2/bson"
+	"sync"
+
+	"github.com/rakyll/magicmime"
```

Comment:
After fiddling around with http.DetectContentType, I found that http.DetectContentType performs vastly worse than magicmime, especially when it comes to executable files (which should be the focus here). For all of my tests magicmime classified them correctly as "application/x-dosexec" or "application/x-executable", while http.DetectContentType just said "application/octet-stream", which is the fallback, if nothing else is recognized. For this reason I would bite the dependency-bullet and settle for magicmime.